okhttp/src/commonMain/kotlin/okhttp3/internal/-ResponseCommon.kt

```diff
@@ -30,9 +31,44 @@ import okhttp3.internal.http.HTTP_MULT_CHOICE
 import okhttp3.internal.http.HTTP_PERM_REDIRECT
 import okhttp3.internal.http.HTTP_SEE_OTHER
 import okhttp3.internal.http.HTTP_TEMP_REDIRECT
-
 import okio.Buffer
 import okio.IOException
+import okio.Source
+import okio.Timeout
+import okio.buffer
+
+class UnreadableResponseBody(
```

Comment:
Should this type be public API, and possible stripBody? for others to use?

How does someone who previously checked the presence of a body before using it (I don't have a convincing example right now, because optionally consuming a body is weird), now check whether it will throw?

Reply:
Yeah very good question. I think it's better without a public API cause we don't want users to think they need to check this.

I think this is similar to the keySet of a MutableMap: you can't insert into it, but it'd be strange to be in a situation where you were even attempting that.

Reply:
Is there a reason you left the type public instead of internal?

Reply:
Nope. Fixed. https://github.com/square/okhttp/pull/7209